test/user.e2e-spec.ts

```diff
@@ -0,0 +1,128 @@
+            database: process.env.TEST_DATABASE_NAME,
+            entities: [User],
+            synchronize: true,
+          }),
+          inject: [ConfigService],
+        }),
+      ],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    app.useGlobalPipes(new ValidationPipe({ whitelist: true, transform: true }));
+    await app.init();
+    userRepository = moduleFixture.get('UserRepository');
+  });
+
+  describe('Create a new user [POST /users]', () => {
+    it('should return a Bad Request 400 error when a user does not send isAdmin in the request', () => {
+      const badRequest = { username: 'Klaven' };
+      return request(app.getHttpServer())
+        .post('/users')
+        .send(badRequest)
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.message[0]).toEqual('isAdmin must be a boolean value');
+        });
+    });
+
+    it('should return a Bad Request 400 error when a user forgets to send username in the request', () => {
+      const badRequest = { isAdmin: true };
+      return request(app.getHttpServer())
+        .post('/users')
+        .send(badRequest)
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.message[0]).toEqual('username should not be empty');
+        });
+    });
+
+    it('should return a Bad Request 400 error when a user sends an empty object', () => {
+      return request(app.getHttpServer())
+        .post('/users')
+        .send({})
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.message[0]).toEqual('username should not be empty');
+          expect(body.message[1]).toEqual('username must be a string');
+          expect(body.message[2]).toEqual('isAdmin must be a boolean value');
+        });
+    });
+
+    it('should create a new user and return the user created', () => {
+      return request(app.getHttpServer())
+        .post('/users')
+        .send(createUser)
+        .expect(201)
+        .then(({ body }) => {
+          expect(body).toEqual({ ...result });
+        });
+    });
+  });
+
+  describe('Get all users [GET /users]', () => {
+    it('should return an array of users', async () => {
+      const { body } = await request
+        .agent(app.getHttpServer())
+        .get('/users')
+        .set('Accept', 'application/json')
+        .expect('Content-Type', /json/)
+        .expect(200);
+
+      expect(body).toEqual([{ ...result }]);
+    });
+  });
+
+  describe('Get one user [GET /user/:id]', () => {
+    it('should return a single user', async () => {
+      const { body } = await request
+        .agent(app.getHttpServer())
+        .get(`/users`)
+        .query({ id: 1 })
+        .set('Accept', 'application/json')
+        .expect(200);
+
+      expect(body).toEqual([{ ...result }]);
+    });
+  });
+
+  afterAll(async () => {
+    //Delete from Test Database when tests are finished
+    await userRepository.query(`DELETE FROM users;`);
+    await app.close();
```

Comment:
This seems like it would get annoying as the number of entities you are dealing with increases. Typeorm has a config option `dropSchema` that you might want to look into.